### DISCUSSION TITLE
Problems with FTPS with CURL

### DISCUSSION THREAD

**Discussion by Q-coder** [curl/curl#11263] [Q&A] [answered]
Hello,
I have two issues both related to FTPS/SFTP transfers using curl. The first one is a 450 error code from Proftpd during the upload.
Example:
curl -k -T MyFile.txt -u xxxUser ftps://xxxServert:990/inbound/MyFile.txt
Enter host password for user 'xxxUser':
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 36.0M    0     0  100 36.0M      0   100M --:--:-- --:--:-- --:--:--  100M
curl: (18) server did not report OK, got 450
Parts of the file is transferred. But it's couple of bytes short.  In this example the source file is 37790596 bytes but the arrived file is : 37437440 bytes.
When I try the same transfer with lftp, it works. No error code returned and the dest file has the same size as the source file.
The lftp command :
lftp -e "debug 10;set ftp:port-ipv4; " -p 990  -u xxxUser:xxxPWD  ftps://xxxServert
put MyFile.txt

---- resetting retry count
copy-peer: EOF on FD 4
copy: get hit eof
copy: waiting for put confirmation
---- Closing data socket
<--- 226 Transfer complete
---> MFMT 20230606183127 MyFile.txt
<--- 213 Modify=20230606183127; MyFile.txt
copy: put confirmed store
copy: get is finished - all done
37790596 bytes transferred
---- Closing idle connection
---> QUIT
<--- 221 Goodbye.
---- Closing control socket
Unfortunately I cannot post all the logfile because of confidential data. 

The other issue is a sftp transfer that writes an empty file on the target system. But with RHEL sftp shell command the same transfer works ok.

curl -k -T myFile.txt  --key xxxmyKeyfile -u xxxUser sftp://xxxServer:2222

myFile is created on the target system but empty. No errors are returned from Proftpd.

sftp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o Port=2222 -o IdentityFile=xxxKeyfile xxxUser@xxxServer
put myFile.txt
quit

myFile.txt is put completely to the target system with the correct size.

What could be the cause of these issues ? 
Curl is curl 7.61.1
lftp is  LFTP | Version 4.8.4 
It's running on RHEL 8.7 

The target system runs CentOS 6.5 (I know it's out of support)
with ProFTPD Version 1.3.3g

Kind Regards,
Gerhard

**Comment by bagder** [comment, marked as answer]
The FTP case is in #11269
